vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -48,7 +63,78 @@ public void setExhibition(JoinPoint jp) {
                         ((Model) obj).addAttribute("publishedSpaces", publishedSpaces);
                     }
                 }
+            }  
+        }
+    }
+
+    @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexOfModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition(); 
+        // If there is no exhibition, we go back to root url page.
+        if(exhibition==null) {
+            return "redirect:/";
+        }
+        //If no exhibition mode has been setup for existing exhibition, we skip modes and aspects.
+        if(exhibition.getMode() == null) {
+            return jp.proceed();
+        }
+        Map<String, String> ids = getIds(args, signature);
+        String spaceId = ids.getOrDefault(IdPrefix.SPACEID.getValue(), "");
+        String moduleId = ids.getOrDefault(IdPrefix.MODULEID.getValue(), "");
+        return redirectRequest(jp, spaceId, moduleId, indexOfModel, exhibition);
+    }
+    
+    // Based on exhibition mode, get the redirect page or pass control to controller.  
+    public Object redirectRequest(ProceedingJoinPoint jp, String spaceId, String moduleId, int modelIndex, Exhibition exhibition) throws Throwable{
+        ISpace space = spaceManager.getSpace(spaceId);
+        IModule module = moduleManager.getModule(moduleId);
+        Object[] args = jp.getArgs();
+        // For exhibition being active, if space or module is not found, show not found page or else give control to controller.
+        if(exhibition.getMode().equals(ExhibitionModes.ACTIVE)) {
+            if(space==null && module==null) {
+                return "notFound";
+            }
+            return jp.proceed();
+        }
+        // If the user is logged in, if space or module invalid, show not found page or else add attribute to show pop up
+        // notifying user about status of exhibition. ModelIndex checks if there is a model to add attribute to.
+        if(authFacade.getAuthenticatedUser()!=null && modelIndex>-1) {
```

Comment:
this should be part of the condition in line 96 (exhiibtion is active or user is authenticated and there is no model).